SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/LogrotateCompressionSettings.java

```diff
@@ -0,0 +1,70 @@
+package com.hubspot.singularity.executor.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+
+public class LogrotateCompressionSettings {
+  private Optional<String> compressCmd = Optional.absent();
+  private Optional<String> uncompressCmd = Optional.absent();
+  private Optional<String> compressOptions = Optional.absent();
+  private Optional<String> compressExt = Optional.absent();
+
+  @JsonCreator
+  public LogrotateCompressionSettings(@JsonProperty("compressCmd") Optional<String> compressCmd,
+                                      @JsonProperty("uncompressCmd") Optional<String> uncompressCmd,
+                                      @JsonProperty("compressOptions") Optional<String> compressOptions,
+                                      @JsonProperty("compressExt") Optional<String> compressExt) {
+    this.compressCmd = compressCmd;
+    this.uncompressCmd = uncompressCmd;
+    this.compressOptions = compressOptions;
+    this.compressExt = compressExt;
+  }
+
+  public static LogrotateCompressionSettings gzip() {
```

Comment:
i'd consider renaming this to `default()`, since it's just an implementation detail of logrotate that it's using `gzip` to compress